issue_drafts/issue_1.md

```diff
@@ -1,3 +1,145 @@
+Try running that script now. I'll wait!
+```
+python 1_fetch/sb_get.py
+```
+Be sure that you're in the correct Conda environment - if not, activate the right environment first:
+```
+conda activate snakemake-tutorial
+```
+
+The script should download a single zip file to `1_fetch/out/pgdl_predictions_04_N45.50-48.00_W92.00-93.00.zip`.
+If the file wasn't downloaded, let your instructor know and we'll get it sorted out!
+
+If the file is there, great!
+Now, let's write a Snakemake rule so that we build in the step of downloading that file to the pipeline.
+
+## Writing a rule
+
+In the root directory of the repository, create a new empty file called `Snakefile` (yes, with the S capitalized).
+Open that file in a text editor, and type
+```
+rule get_sb_data:
+```
+That's how you define a Snakemake rule.
+> NOTE: It's a lot like the way functions get defined in Python, except instead of `def` we type `rule`.
+> In fact, Snakefiles are basically Python files with some added functionality.
+> You can include Python code directly in a Snakefile - define variables, or functions, or anything you want - and that code will work the same as if it were in a `.py` file.
+
+Next, we'd usually list the files that are inputs to this rule.
+However, since this is the first rule of our pipeline, we don't have any inputs.
+So, let's skip straight to outputs.
+We'll use the output directive to define the outputs of this rule.
+In this case, there's only one output: the zip file.
+Easy enough!
+Let's add this to our rule:
+```
+rule get_sb_data:
+    output:
+        1_fetch/out/pgdl_predictions_04_N45.50-48.00_W92.00-93.00.zip
+```
+
+Finally, we need to tell Snakemake how to create this zip file - what code to run.
+There are a few ways to do this, but we'll start with our favorite way: the `script` directive.
+The `script` directive tells Snakemake to run a script in order to make the output file(s) for this rule.
+In this case, we know that all we need to do is to run `1_fetch/sb_get.py` to download the file.
+So, let's specify that Python file as our script.
+```
+rule get_sb_data:
+    output:
+        1_fetch/out/pgdl_predictions_04_N45.50-48.00_W92.00-93.00.zip
+    script:
+        1_fetch/sb_get.py
+```
+Like with any Python file, be sure to pay attention to indentation!
+Since the script is a `.py`, Snakemake knows to use `python` to run it.
+Unless we specify otherwise, the version of Python that's used to run the script will be the version that's called when you type `python` into the command line.
+So, when you want to run the pipeline, be sure that you're in the correct Conda environment.
+
+Okay, we've got our Snakefile written with its first rule!
+Let's try it out.
+
+## Executing the pipeline
+
+First, go ahead and delete the zip file we downloaded earlier - the one that's in `1_fetch/out/`.
+Next, execute our new rule using the following command:
+```
+snakemake --cores 1 1_fetch/out/pgdl_predictions_04_N45.50-48.00_W92.00-93.00.zip
```

Comment:
I honestly haven't done much rule-by-rule running of snakemake. It seems easier to type the rule name than to remember what the output file would be and type it. Should we suggest that instead? I don't have a hard stance...I'm curious if you think one is better than the other.
```suggestion
snakemake --cores 1 -R get_sb_data
```

Reply:
If a rule's output has wildcards in it, then the only way to run that rule is to specify its output. For that reason, I've become accustomed to specifying output files instead of rule names. Using the output file as an argument works in most if not all cases. My leaning is to demonstrate this practice first so that folks "imprint" on it.